nixos/doc/manual/configuration/firewall.xml

```diff
@@ -25,8 +25,21 @@ networking.firewall.allowedTCPPorts = [ 80 443 ];
 Note that TCP port 22 (ssh) is opened automatically if the SSH daemon
 is enabled (<option>services.openssh.enable = true</option>).  UDP
 ports can be opened through
-<option>networking.firewall.allowedUDPPorts</option>.  Also of
-interest is
+<option>networking.firewall.allowedUDPPorts</option>.
+
+To open ranges of TCP ports:
+
+<programlisting>
+networking.firewall.allowedTCPPortRanges = [
+  { from = 4000; to = 4007; }
+  { from = 8000; to = 8010; }
+];
+</programlisting>
+
+Similarly, UDP port ranges can be opened through
+<option>networking.firewall.allowedUDPPortRanges</option>.
+
+Also of interest is
```

Comment:
Note that this isn't markdown but docbook, so en emtpy line will not produce a new paragraph. You need to close the para tag and open a new one.

Reply:
Better?